src/Hedgehog/Extras/Test/File.hs

```diff
@@ -170,7 +174,9 @@ openFile filePath mode = GHC.withFrozenCallStack $ do
 readFile :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m String
 readFile filePath = GHC.withFrozenCallStack $ do
   void . H.annotate $ "Reading file: " <> filePath
-  H.evalIO $ IO.readFile filePath
+  liftIO $ IO.withFile filePath IO.ReadMode $ \handle -> do
```

Comment:
`evalIO` is needed for proper reporting of IO exceptions
```suggestion
  H.evalIO $ IO.withFile filePath IO.ReadMode $ \handle -> do
```

Why this change only on two functions? I think the change should apply to all functions using string / text when accessing files:
`appendFile`, `textWriteFile`, `textReadFile`, `fileContains`

Reply:
I think you are right.  However, it makes sense to accept this PR as an improvement to the library and do the other fixes in a future PR.

Reply:
>do the other fixes in a future PR

Fine but let's do them before making a new release... And the version of hedgehog-extras with this change was tagged already. Can we try to limit technical debt we're introducing?

Or at least put an effort to make technical debt manageable and traceable by creating follow-up tickets/prs?